integration-tests/app_id_test.go

```diff
@@ -7,110 +7,116 @@ import (
-	errStop := BuildStopContainers()
-	if errStop != nil {
-		fmt.Printf("Error stopping : %v", errStop)
-	}
+	assert.Equal(suite.T(), 403, res.StatusCode, "Invalid http response code for appId forbidden test: %d", res.StatusCode)
 }
 
-func TestUserKeySuccess(t *testing.T) {
-	err := StartContainers("./configs/app-id/docker-compose.yaml")
-	// Sleep required in github actions to prevent failing of tests in case of services getting late to start.
-	time.Sleep(10 * time.Second)
-	if err != nil {
-		fmt.Printf("Error: %v", err)
-		t.Fatalf("Error starting docker-compose: %v", err)
-	}
+func (suite *AppCredentialTestSuite) TestUserKeySuccess() {
 	client := &http.Client{}
 	req, errReq := http.NewRequest("GET", "http://127.0.0.1:9095/", nil)
-	if errReq != nil {
-		fmt.Printf("Error: %v", errReq)
-		StopContainers("./configs/app-id/docker-compose.yaml")
-		t.Fatalf("Error creating http request: %v", errReq)
-	}
+	require.Nilf(suite.T(), errReq, "Error creating the HTTP request: %v", errReq)
 	q := req.URL.Query()
-	q.Add("api_key", "46de54605a1321aa3838480c5fa91bcc")
+	q.Add("api_key", "04fa57ba1e465fb53fddc21ead8a7fc0")
 	req.URL.RawQuery = q.Encode()
 	res, errHTTP := client.Do(req)
-	if errHTTP != nil {
-		fmt.Printf("Error: %v", errHTTP)
-		StopContainers("./configs/app-id/docker-compose.yaml")
-		t.Fatalf("Error sending http request: %v", errHTTP)
-	}
+	require.Nilf(suite.T(), errHTTP, "Error sending the HTTP request: %v", errHTTP)
 	fmt.Printf("Response: %v", res)
-	assert.Equal(t, 200, res.StatusCode, "Invalid http response code for user_key success test: %v", res.StatusCode)
-	assert.Equal(t, "3scale", res.Header.Get("Powered-By"), "Inavlid http header for powered-by: %v", res.Header.Get("Powered-By"))
-	errStop := StopContainers("./configs/app-id/docker-compose.yaml")
-	if errStop != nil {
-		fmt.Printf("Error stopping : %v", errStop)
-	}
+	assert.Equal(suite.T(), 200, res.StatusCode, "Invalid http response code for user_key success test: %v", res.StatusCode)
+}
+
+func (suite *AppCredentialTestSuite) TestUserKeyForbidden() {
+	client := &http.Client{}
+	req, errReq := http.NewRequest("GET", "http://127.0.0.1:9095/", nil)
+	require.Nilf(suite.T(), errReq, "Error creating the HTTP request: %v", errReq)
+	q := req.URL.Query()
+	q.Add("api_key", "04fa57ba1e465fb53fddc21ead8a7fc1")
+	req.URL.RawQuery = q.Encode()
+	res, errHTTP := client.Do(req)
+	require.Nilf(suite.T(), errHTTP, "Error sending the HTTP request: %v", errHTTP)
+	fmt.Printf("Response: %v", res)
+	assert.Equal(suite.T(), 403, res.StatusCode, "Invalid http response code for user_key forbidden test: %v", res.StatusCode)
 }
+
+// func (suite *AppCredentialTestSuite) TestUnlimitedUserKey() {
+// 	client := &http.Client{}
+// 	req, errReq := http.NewRequest("GET", "http://127.0.0.1:9095/", nil)
+// 	require.Nilf(suite.T(), errReq, "Error creating the HTTP request: %v", errReq)
+// 	q := req.URL.Query()
+// 	q.Add("api_key", "10e81d5c065a537b05ab7d78a7156fc5")
```

Comment:
Is this part of the WIP?

Reply:
To support unlimited applications, few changes are required in the cache filter side. So just commented out until it gets fixed. Can remove if this looks ugly

Reply:
No, keep it, but either make sure it compiles or if you comment it out then add a comment on top to know why.